allennlp/semparse/worlds/wikitables_variable_free_world.py

```diff
@@ -147,34 +147,52 @@ def get_agenda(self):
         agenda_items = []
         question_tokens = [token.text for token in self.table_context.question_tokens]
         question = " ".join(question_tokens)
+
+        if "at least" in question:
+            agenda_items.append("filter_number_greater_equals")
+        if "at most" in question:
+            agenda_items.append("filter_number_lesser_equals")
+
+        comparison_triggers = ["greater", "larger", "more"]
+
+        if any("no %s than" %word in question for word in comparison_triggers):
+            agenda_items.append("filter_number_lesser_equals")
+        elif any("%s than" %word in question for word in comparison_triggers):
+            agenda_items.append("filter_number_greater")
+
+
         for token in question_tokens:
             if token in ["next", "after", "below"]:
                 agenda_items.append("next")
             if token in ["previous", "before", "above"]:
                 agenda_items.append("previous")
+
+
             if token == "total":
                 agenda_items.append("sum")
             if token == "difference":
                 agenda_items.append("diff")
             if token == "average":
                 agenda_items.append("average")
-            if token in ["least", "top", "smallest", "shortest", "lowest"]:
+
+
+            if token in ["least", "smallest", "shortest", "lowest"] and "at least" not in question:
                 # This condition is too brittle. But for most logical forms with "min", there are
                 # semantically equivalent ones with "argmin". The exceptions are rare.
                 if "what is the least" in question:
                     agenda_items.append("min")
                 else:
                     agenda_items.append("argmin")
-            if token in ["most", "largest", "highest", "longest", "greatest"]:
+            if token in ["most", "largest", "highest", "longest", "greatest"] and "at most" not in question:
                 # This condition is too brittle. But for most logical forms with "max", there are
                 # semantically equivalent ones with "argmax". The exceptions are rare.
                 if "what is the most" in question:
                     agenda_items.append("max")
                 else:
                     agenda_items.append("argmax")
-            if token == "first":
+            if token in ["first", "top"]:
                 agenda_items.append("first")
-            if token == "last":
+            if token == ["last", "bottom"]:
```

Comment:
The test is failing because of this line. `==` -> `in`.

Reply:
got it